components/script/dom/webglrenderingcontext.rs

```diff
@@ -177,6 +177,44 @@ impl TextureUnitBindings {
     }
 }
 
+#[derive(JSTraceable)]
+pub struct WebGLRenderingContextLimits {
+    pub max_vertex_attribs: u32,
+    pub max_vertex_uniform_vectors: u32,
+    pub max_varying_vectors: u32,
+    pub max_vertex_texture_image_units: u32,
+    pub max_combined_texture_image_units: u32,
+    pub max_texture_image_units: u32,
+    pub max_fragment_uniform_vectors: u32,
+    pub max_tex_size: u32,
+    pub max_cube_map_tex_size: u32,
+}
+
+impl WebGLRenderingContextLimits {
+    pub fn new(renderer: &WebGLMsgSender) -> WebGLRenderingContextLimits {
+        let get_param = |param| {
+            let (sender, receiver) = webgl_channel().unwrap();
+
+            renderer.send(WebGLCommand::GetParameter(param, sender)).unwrap();
+            match receiver.recv().unwrap() {
+                Ok(WebGLParameter::Int(x)) => x as u32,
+                _ => 0,
+            }
+        };
+
+        WebGLRenderingContextLimits {
+            max_vertex_attribs: get_param(constants::MAX_VERTEX_ATTRIBS),
```

Comment:
Could you initialize this struct in the WebGLThread when a context is created?. I'd like to also avoid the getter channel steps when creating the context:  you have direct access to the context.gl() there in webgl_thread.rs. And send the result where the previous `limits` was used

Reply:
I'm trying, but it's such a layering violation that I'm having a hard time.

Reply:
You can define the struct in webgl.rs (canvas_traits component). That way it can be used in both layers without problems.

Reply:
The problem was then trying to get the constants back from the script component.  gleam::gl to the rescue.